webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/yaml/YamlTechnique.scala

```diff
@@ -222,42 +263,23 @@ class YamlTechniqueSerializer(resourceFileService: ResourceFileService) {
         )
     }
   }
+}
 
-  def toReportingLogic(reporting: Reporting): PureResult[ReportingLogic] = {
-    ReportingLogic.parse(reporting.mode ++ (reporting.id.map(":" ++ _).getOrElse("")))
-  }
-  private def toMethodElem(item: MethodItem): PureResult[MethodElem]     = {
-    item.items match {
-      case Some(items) =>
-        for {
-          reporting <- item.reporting.map(toReportingLogic).getOrElse(Right(WeightedReport))
-          items     <- items.accumulatePure(toMethodElem)
-        } yield {
-          MethodBlock(
-            item.id,
-            item.name,
-            reporting,
-            item.condition.getOrElse(""),
-            items
-          )
-        }
-      case None        =>
-        item.method match {
-          case Some(method) =>
-            Right(
-              MethodCall(
-                BundleName(method),
-                item.id,
-                item.params.getOrElse(Map()),
-                item.condition.getOrElse(""),
-                item.name,
-                item.reporting == Some("disabled")
-              )
-            )
-          case None         => Left(Consistancy("error"))
+class YamlTechniqueSerializer(resourceFileService: ResourceFileService) {
+  import YamlTechniqueSerializer._
 
-        }
+  // shortcut to avoid having to import yamlOps / toIO, but can be avoided
+  def toYml(technique: Technique): IOResult[String] = technique.toYaml().toIO
```

Comment:
Yep, I'd like to remove them as i think they may be useless. maybe they are useful in case of classes and services that mix yaml and json and you need to import nearer to the use case and in this case we could keep a global json import and use the service function